### PR TITLE
20616 add mobile shipment customer flows main

### PR DESCRIPTION
**Pull request by KonstanceH**
## Summary

Integration PR: https://github.com/transcom/mymove/pull/13559/files

REBASE INT PR: https://github.com/transcom/mymove/pull/13770

**Note**
RequiredTag has been removed from this PR and will be added in from 19226 Main PR
**MobileHomeShipmentCard.test.jsx:**
This PR adds a new line check for tests that are going to be added to integration
from this PR: https://github.com/transcom/mymove/pull/13767
And also ups the version for client/server test coverage to match that intergration PR as well.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. Login as a customer
2. Create a move
3. Add a mobile home shipment
4. Confirm the shipment has been added to the list with the right styling
5. Confirm ability to delete and edit your added mobile home shipment(s).

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
_**Mobile Home added to shipment options**_
Screenshot 2024-08-26 at 5 50 06 AM

_**Mobile Home Info Page**_
Screenshot 2024-08-26 at 5 50 13 AM

_**Incomplete Mobile Home added to shipment list**_
Screenshot 2024-08-27 at 3 32 48 PM

### CODE REVIEW

File: src/shared/styles/colors.scss [hunk 1]
```diff
@@ -38,7 +38,8 @@ $accent-ub: #f2938c;
 $accent-nts: #d85bef;
 $accent-ntsr: #8168b3;
 $accent-boat: #5d92ba;
-$accent-pro-gear: #71767a;
+$accent-mobile-home: #1fd819;
+$accent-pro-gear: #71767A;
```
[the remark this answers is not in the document]
Comment (KonstanceH) [reply src/shared/styles/colors.scss]: ah okay just pushed up the change

**Review by taeJungCaci — APPROVED** [review]
LGTM. Only difference was in `src/components/Customer/MobileHomeShipment/MobileHomeShipmentForm/MobileHomeShipmentForm.jsx` related to `RequiredTag` which we discussed and is due to merge conflicts. This diff will be added in https://github.com/transcom/mymove/pull/13717 where it belongs.

**Review by brooklyn-welsh — APPROVED** [review]
LGTM!

**Comment by taeJungCaci** [comment]
Looks like I can't merge due to coverage issue. You may have to include tests for mobile home shipment card that you have in this [INT PR for B-20653](https://github.com/transcom/mymove/pull/13620/files#diff-8179b1adba6a1d6455d88709a61b32ff1a04b9c95c5800b249afe578c3601ce3). Or figure out how to bypass this coverage issue.

**Comment by KonstanceH** [comment]
> Looks like I can't merge due to coverage issue. You may have to include tests for mobile home shipment card that you have in this [INT PR for B-20653](https://github.com/transcom/mymove/pull/13620/files#diff-8179b1adba6a1d6455d88709a61b32ff1a04b9c95c5800b249afe578c3601ce3). Or figure out how to bypass this coverage issue.

Okay I just added in the tests from B-20653 as you suggested hopefully it gets past the coverage tests

**Comment by taeJungCaci** [comment]
> > Looks like I can't merge due to coverage issue. You may have to include tests for mobile home shipment card that you have in this [INT PR for B-20653](https://github.com/transcom/mymove/pull/13620/files#diff-8179b1adba6a1d6455d88709a61b32ff1a04b9c95c5800b249afe578c3601ce3). Or figure out how to bypass this coverage issue.
> 
> Okay I just added in the tests from B-20653 as you suggested hopefully it gets past the coverage tests

Thanks! But I was actually referring to the Jest test `src/components/Customer/Review/ShipmentCard/MobileHomeShipmentCard/MobileHomeshipmentCard.test.jsx` because I don't think the playwright test are included in the coverage. Not sure, but the jest file probably belongs on this PR anyways for the card component.

**Comment by KonstanceH** [comment]
> > > Looks like I can't merge due to coverage issue. You may have to include tests for mobile home shipment card that you have in this [INT PR for B-20653](https://github.com/transcom/mymove/pull/13620/files#diff-8179b1adba6a1d6455d88709a61b32ff1a04b9c95c5800b249afe578c3601ce3). Or figure out how to bypass this coverage issue.
> > 
> > 
> > Okay I just added in the tests from B-20653 as you suggested hopefully it gets past the coverage tests
> 
> Thanks! Sorry for the confusion but I was actually referring to the Jest test and the link should have directed you to the file`src/components/Customer/Review/ShipmentCard/MobileHomeShipmentCard/MobileHomeshipmentCard.test.jsx` because I don't think the playwright test are included in the coverage - not sure. But the jest file probably belongs in this PR anyways for the card component.

Ah okay I just pushed up those tests

**Comment by taeJungCaci** [comment]
Looks like those tests weren't enough to make the coverage fail go away. I guess you have to write tests for some of the listed files in the coverage test? Possibly `pkg/handlers/internalapi/moves.go`? Since the code changes you've made in there are feature flag related the test would need that flag also fyi.

**Comment by taeJungCaci** [comment]
> Looks like those tests weren't enough to make the coverage fail go away. I guess you have to write tests for some of the listed files in the coverage test? Possibly `pkg/handlers/internalapi/moves.go`right? Since the code changes you've made in there are feature flag related the test would need that flag also fyi.

Scratch what I just said. I think they changed it so I can still merge in with the coverage fail?

**Review by taeJungCaci — APPROVED** [review]
MobileHomeShipmentCard.test.jsx has been added to this PR for coverage. This test file had been added in this PR in INT: [20653](https://github.com/transcom/mymove/pull/13620/files#diff-8179b1adba6a1d6455d88709a61b32ff1a04b9c95c5800b249afe578c3601ce3)

**Pull request merged**